app/src/main/java/capsicum/game/wordfox/IVmethods.java

```diff
@@ -3,44 +3,27 @@
 import android.content.Context;
 import android.graphics.BitmapFactory;
 import android.graphics.Point;
-import android.graphics.Rect;
 import android.view.Display;
+import android.view.View;
 import android.view.WindowManager;
-import android.widget.ImageView;
 import android.widget.TextView;
 
+/**
+ * Created by Desmond
+ */
 public class IVmethods {
 
-    public static void setTVwidthPercentOfIV(ImageView instructionFoxSpeechBubbleIV,
-                                             TextView instructionFoxTV, double percent, int stringResId) {
-
-        int[] xy1 = new int[3];
-        Rect myRect = new Rect();
-
-        instructionFoxSpeechBubbleIV.post(new Runnable() {
+    // Observe one view and use it's laid out width to decide the width of a difference view.
+    public static void setWidthAsPercentOfLaidOutView(View viewToObserve, TextView viewToModify, double percent) {
+        viewToObserve.post(new Runnable() {
             @Override
             public void run() {
-                instructionFoxSpeechBubbleIV.getGlobalVisibleRect(myRect);
-                xy1[2] = myRect.width();
-
-                if (stringResId != 0){
-                    instructionFoxTV.setText(stringResId);
-                }
-
-                instructionFoxTV.requestLayout();
-                instructionFoxTV.setWidth((int) (xy1[2] * percent));
+                viewToModify.setWidth((int) (viewToObserve.getWidth() * percent));
             }
         });
-
-    }
-
-
-    public static void setTVwidthPercentOfIV(ImageView instructionFoxSpeechBubbleIV,
-                                             TextView instructionFoxTV, double percent, String myString) {
-        setTVwidthPercentOfIV(instructionFoxSpeechBubbleIV, instructionFoxTV, percent, 0);
-        instructionFoxTV.setText(myString);
```

Comment:
Removed setText because seems cleaner to keep a single responsibility for this function. You can setText() on the original TextView in just a single line so there's no saving by adding it here. @desseepe